tests/common/semantics_by_value.h

```diff
@@ -30,49 +30,87 @@
 namespace common_by_value_semantics {
 
 /**
- * @brief Check equality-comparable operations on the host side
+ * @brief Provides a safe index for checking an operation
  */
-template <typename T>
-void check_on_host(sycl_cts::util::logger& log, const T& a,
-                   const std::string& testName) {
+enum class current_check : size_t {
+  reflexivity_equal_self,
+  reflexivity_not_equal_self,
+  equal_copy,
+  equal_copy_symmetry,
+  not_equal_copy,
+  not_equal_copy_symmetry,
+  transitivity_equal,
+  transitivity_not_equal,
+  equal_other,
+  equal_other_symmetry,
+  not_equal_other,
+  not_equal_other_symmetry,
+  SIZE  // This should be last
```

Comment:
The uppercase is strange regarding the coding style since it is not a macro.
Or at least I hope this is not a macro, by the way... :-)